node/caching-compiler.js

```diff
@@ -70,7 +71,7 @@ function readWithCache(info, content) {
   };
 
   if (reify && reify.parser) {
-    compileOptions.parse = require(reify.parser).parse;
+    compileOptions.parse = utils.require(reify.parser).parse;
```

Comment:
Come to think of it, could we just use `module.require` here (and other places where `utils.require` is used)? That way, module resolution would be relative to `module.id` rather than always `lib/utils.js`.